src/stdlib/SDL_memset.c

```diff
@@ -0,0 +1,135 @@
+/*
+  Simple DirectMedia Layer
+  Copyright (C) 1997-2024 Sam Lantinga <slouken@libsdl.org>
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely, subject to the following restrictions:
+
+  1. The origin of this software must not be misrepresented; you must not
+     claim that you wrote the original software. If you use this software
+     in a product, an acknowledgment in the product documentation would be
+     appreciated but is not required.
+  2. Altered source versions must be plainly marked as such, and must not be
+     misrepresented as being the original software.
+  3. This notice may not be removed or altered from any source distribution.
+*/
+#include "SDL_internal.h"
+
+
+#ifdef SDL_memset
+#undef SDL_memset
+#endif
+#if SDL_DYNAMIC_API
+#define SDL_memset SDL_memset_REAL
+#endif
+void *SDL_memset(SDL_OUT_BYTECAP(len) void *dst, int c, size_t len)
+{
+#ifdef HAVE_MEMSET
+    return memset(dst, c, len);
```

Comment:
Can we use `__builtin_memset` here as you did in `SDL_memcpy`?
```suggestion
#ifdef __GNUC__
    return __builtin_memset(dst, c, len);
#elif defined(HAVE_MEMSET)
    return memset(dst, c, len);
```